### ISSUE TITLE
no-nil false positive on switch statements

### ISSUE THREAD

**Issue by blackxored** [jfmengels/eslint-plugin-fp#36]
Take this function:

```javascript
const someFunction = (action) => {
  switch (action.type) {
    case 'SOME_ACTION': {
      return assoc('showNetworkIndicator', true, state);
    }
    default: {
      return state;
    }
  }
}
```
`fp/no-nil` complains about it. Am I missing something or this should not have an error on it?

**Comment by carmanchris31** [comment]
See https://github.com/jfmengels/eslint-plugin-fp/issues/14
